docs/support/tutorials/nextflow-puhti.md

```diff
@@ -1,191 +1,309 @@
-
 # Running Nextflow pipelines on Puhti
-[Nextflow](https://www.nextflow.io/) is a scalable and reproducible scientific workflow 
-management system that interacts with containerised applications to perform 
-compute-intensive tasks. Nextflow provides built-in support for containers 
-like Singularity, which is a containerisation technology that is compliant 
-with HPC environment. Although Nextflow pipelines allow us to choose Docker 
-engine as an executor for running pipelines, please note that Docker 
-containers can't be used in Puhti due to the lack of administrative privileges for researchers.
+
+[Nextflow](https://www.nextflow.io/) is a scalable and reproducible scientific
+workflow management system that interacts with containerized applications to
+perform compute-intensive tasks. Nextflow provides built-in support for
+HPC-friendly containers such as Apptainer and Singularity. Although Nextflow
+pipelines allow us to choose Docker engine as an executor for running
+pipelines, please note that Docker containers can't be used on Puhti due to the
+lack of administrative privileges for regular users.
 
 ## Strengths of Nextflow
 
 * Easy installation
-* No database needed
-* Implicit parallelism supported
+* Supports implicit parallelism
 * Can handle complex dependencies and conditional execution
-* Supports error recovery
+* Handles error recovery
 
 ## Disadvantages of Nextflow
 
-* Limited MPI support
-* May create a lot of job steps and excess IO
-* Does not integrate well with Slurm
+* Has limited MPI support
+* Creates a lot of job steps and excessive I/O load
+* Does not efficiently integrate with Slurm
 
-## How Singularity works with Nextflow
-The integration of Singularity containers with Nextflow follows the same 
-execution model as with Docker containers. So it does not require additional 
-modifications in a Nextflow script as such to be compatible with Singularity. 
-One has to just enable the use of Singularity engine (instead of Docker) 
-in the Nextflow configuration file. Nextflow is able to pull remote container 
-images stored in Singularity or Docker-Hub registry. The remote container 
-images are usually specified in your Nextflow script or configuration 
-file by simply prefixing the image name with the `shub://` or `docker://`. 
-It is also possible to specify a different Singularity image for each 
-process definition in the Nextflow pipeline script. 
+## Use Apptainer/Singularity containers with Nextflow
 
-Here is a generic recipe for running your Nextflow pipeline on Puhti :
+Containers can be smoothly integrated with Nextflow pipelines. No additional
+modifications to Nextflow scripts are needed except enabling the Singularity
+engine (instead of Docker) in the Nextflow configuration file in the HPC
+environment. Nextflow is able to pull remote container images stored in
+Singularity or Docker Hub registry. The remote container images are usually
```

Comment:
Should it be singularity or apptainer?

Reply:
As this is general description, we can mention both alternatives (Singularity and Apptainer).